### ISSUE TITLE
Help for a use case of Multiple Losses

### ISSUE THREAD

**Issue by rangwani-harsh** [f-dangel/backpack#69]
Hi,
Thanks for the awesome library!!
I have a use case in which I have multiple loss functions on which I have to call backward without using any reduction like mean or sum. I want to calculate gradients for different losses parallely. 
```
losses = [loss1, loss2, loss3]
losses.backward()

print(param.grad)
## It should contain the jacobian of the gradients

```
As it's possible to calculate gradients wrt every sample in a batch (I don't want gradients for each sample) is it possible to generalize to this use case?

**Comment by fKunstner** [comment]
Hi Harsh, thanks for reaching out. 

If I get it, you'd like a a general purpose jacobian functionality? 

That's interesting feature request, but I don't think we can make backpack do that in the general case with a few lines of code. I'd recommend looking at other options if you need this functionality quickly.

Out of curiosity, what's the use case here? It's unlikely, but maybe we can recommend something if the use case is small enough.

Cheers

**Comment by rangwani-harsh** [comment]
Hi,
Thanks for your response
 I was trying to implement the PCGrad paper which is released on arxiv recently. They show great results on multitask optimization. The original repo with tf implementation is here https://github.com/tianheyu927/PCGrad. What I actually want is to implement this particular line in parallel https://github.com/tianheyu927/PCGrad/blob/c5fbd7c856526373828074f06875230f7f3ee79e/PCGrad_tf.py#L39. 

My pytorch basic implementation is here:
https://github.com/rangwani-harsh/PC_Grad_Pytorch/blob/f14f265dcc3f10f1ea0f39836a259e31ed1c7d6a/pc_grad_pytorch.py#L18

Thanks

**Comment by fKunstner** [comment]
Thanks for the details. As expected, backpack's not going to be of much help, especially if you want to use specialized models (I took a look at [routing networks](https://github.com/cle-ros/RoutingNetworks)).

The best way to implement this would require a Jacobian-matrix product, but PyTorch only offers Jacobian-vector product and there are no easy way to vectorize it. There are some interesting moves in that direction ([dynamic batching](https://github.com/pytorch/pytorch/issues/1642), [nested tensors](https://github.com/pytorch/nestedtensor), [vmap proof of concept](https://github.com/pytorch/pytorch/pull/32836)), but nothing we can use at the moment as far as I know.

If you're set on PyTorch, [multiprocessing](https://github.com/pytorch/pytorch/issues/33937) might help, otherwise [JAX](https://jax.readthedocs.io/en/latest/notebooks/autodiff_cookbook.html#Jacobian-Matrix-and-Matrix-Jacobian-products) might be an option.

**Comment by rangwani-harsh** [comment]
Thanks for the detailed answer.
